internal/k8s/app_protect_configuration_test.go

```diff
@@ -0,0 +1,1126 @@
+package k8s
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func parseTime(value string) *time.Time {
+	t, err := time.Parse(timeLayout, value)
+	if err != nil {
+		panic(err)
+	}
+
+	return &t
+}
+
+func sliceCmpFunc(x, y *unstructured.Unstructured) bool {
+	return x.GetUID() > y.GetUID()
+}
+
+var unstructuredSliceCmpOpts = cmpopts.SortSlices(sliceCmpFunc)
```

Comment:
Could you possibly clarify why this is needed? I tried removing it and the tests still passed.

Reply:
Sure
It passed for me as well, so i pushed and got the errors. when rerunning localy and disabling cache i saw one in 10 tries fail, due to the slices being in incorect order.
This additional option sorts the *unstructured.Unstructured slices in udschanges to prevent this. Reran ~20 times after this change and had 100% sucess

Reply:
oh, got it. thanks for clarification 👍 
that's probably because that list is constructed here https://github.com/nginxinc/kubernetes-ingress/blob/24ed921e3f1e136d407f0afcfa824506b7d03f56/internal/k8s/app_protect_configuration.go#L431 from a map. However, when iterating a map, the keys are not guaranteed to be returned in the same order. 

In a similar type, we sort the keys when iterating over the map -- https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/k8s/configuration.go#L549

updating the test to use a custom comparator looks good to me, as long as the different doesn't cause any issues when IC is running

Reply:
right, the order maters only in tests so i thought i fix it there.